CC: @cfriedt

```
/__w/zephyr/zephyr/lib/posix/pthread.c: In function 'pthread_getconcurrency':
/__w/zephyr/zephyr/lib/posix/pthread.c:477:16: error: 'ret' may be used uninitialized [-Werror=maybe-uninitialized]
  477 |         return ret;
      |                ^~~
/__w/zephyr/zephyr/lib/posix/pthread.c:471:13: note: 'ret' was declared here
  471 |         int ret;
```

--- a/drivers/ethernet/eth_sam_gmac.c
+++ b/drivers/ethernet/eth_sam_gmac.c
@@ -2439,7 +2439,7 @@ static int ptp_clock_sam_gmac_adjust(const struct device *dev, int increment)
 	const struct eth_sam_dev_cfg *const cfg = ptp_context->eth_dev->config;
 	Gmac *gmac = cfg->regs;
 
-	if ((increment <= -NSEC_PER_SEC) || (increment >= NSEC_PER_SEC)) {
+	if ((increment <= -(int)NSEC_PER_SEC) || (increment >= (int)NSEC_PER_SEC)) {
 		return -EINVAL;
 	}
 

